### DISCUSSION TITLE
[VST2 plugin] Validation error of VST2 in Logic

### DISCUSSION THREAD

**Discussion by dobo365** [RomanKubiak/ctrlr#605] [Q&A]
Hi,
   some people are experiencing a validation error in Logic.
The error message is saying many times
"WARNING: retrievedValue = 0.000000 (was 1.000000), Parameter did not retain maximum value when set - ERROR: Parameter did not retain set value when Initialized"

Then after restarting Logic (once or a few times), it suddenly appears as OK.

Does anyone have an idea of what it could be?

**Comment by Ted-juh** [comment]
It's nice of Logic to inform you about a parameter doing bad things, but not tell which one. Logic seems to test the parameters and detects when values are going "out of bounds" or "not the same as before". 

Could be a parameter getting its value from another one, or just one parameter that decides to go bonkers when normalizing/denormalizing. Or setting a true/ false the wrong way. Hard to tell with just "0.0000 and 1.00000". 

Are you initializing parameters on startup to be set to a certain value?

**Comment by Godlike-Productions** [comment]
We've had this issue with the Kurzweil Remote plugins.

As best we can see this is related to a component setting the value of another component, and handling in the value tree. There was also a bug in the validation tool that if even 1 component affected even 1 other component, every component (including the name of the plugin) had to be cast in a particular way, even all the hidden stuff, and menu bars from the editor etc.

For MIDI plugins, it really doesn't make much sense if you retrieve any values from a synth, as the values are not kept in the plugin.  There are also potential issues now with the logic .  Lots of people are having issues with the logic validation tool.  It would be awesome if this could be fixed, but I have no idea about how to approach it.
